internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/helix/TestControllerKMERegistrationFromMessageHeader.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class TestControllerKMERegistrationFromMessageHeader {
+  private static final Logger LOGGER = LogManager.getLogger(TestControllerKMERegistrationFromMessageHeader.class);
+  private static final int TEST_TIMEOUT = 90_000; // ms
+  private static final int NUMBER_OF_CHILD_DATACENTERS = 1;
+  private static final int NUMBER_OF_CLUSTERS = 1;
+
+  private static final String[] CLUSTER_NAMES =
+      IntStream.range(0, NUMBER_OF_CLUSTERS).mapToObj(i -> "venice-cluster" + i).toArray(String[]::new);
+
+  private List<VeniceMultiClusterWrapper> childDatacenters;
+  private VeniceTwoLayerMultiRegionMultiClusterWrapper multiRegionMultiClusterWrapper;
+
+  @BeforeClass
+  public void setUp() {
+    Properties controllerProps = new Properties();
+    controllerProps.put(DEFAULT_NUMBER_OF_PARTITION_FOR_HYBRID, 2);
+    controllerProps.put(DEFAULT_MAX_NUMBER_OF_PARTITIONS, 3);
+    controllerProps.put(DEFAULT_PARTITION_SIZE, 1024);
+    controllerProps.put(KME_REGISTRATION_FROM_MESSAGE_HEADER_ENABLED, true);
+    multiRegionMultiClusterWrapper = ServiceFactory.getVeniceTwoLayerMultiRegionMultiClusterWrapper(
+        NUMBER_OF_CHILD_DATACENTERS,
+        NUMBER_OF_CLUSTERS,
+        1,
+        3,
+        1,
+        1,
+        1,
+        Optional.of(controllerProps),
+        Optional.of(controllerProps),
+        Optional.empty());
+
+    childDatacenters = multiRegionMultiClusterWrapper.getChildRegions();
+  }
+
+  @AfterClass(alwaysRun = true)
+  public void cleanUp() {
+    Utils.closeQuietlyWithErrorLogged(multiRegionMultiClusterWrapper);
+  }
+
+  @Test(timeOut = TEST_TIMEOUT)
+  public void testKMERegistrationThroughAdminTopicChannel() {
+    String clusterName = CLUSTER_NAMES[0];
+    String storeName = Utils.getUniqueString("store");
+
+    VeniceControllerWrapper pController =
+        multiRegionMultiClusterWrapper.getLeaderParentControllerWithRetries(clusterName);
+
+    VeniceMultiClusterWrapper child = childDatacenters.get(0);
+
+    // Remove the latest schema from child controller's local value serializer and remove it from child colo's schema
+    // repository (ZK).
+    VeniceControllerWrapper leaderController = child.getLeaderController(clusterName);
+    KafkaValueSerializer valueSerializer =
+        leaderController.getController().getVeniceControllerService().getKafkaValueSerializer();
+    TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, true, () -> {
+      Assert.assertTrue(
+          valueSerializer.getProtocolVersionSize() >= AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE
+              .getCurrentProtocolVersion());
```

Comment:
Just curious, why do we need this assertion?

Reply:
Sure, it does no harm to add additional assertion if you think it will reduce the chance of race. Though I thought local protocol loading (i.e. add schemas from local resource into `protocolVersionToReader`) happens in `InternalAvroSpecificSerializer` constructor. Does `ClusterLeaderInitializationManager.execute` also trigger local protocol loading?

Reply:
Oh, I see the problem. I actually wanted to wait the schema to appear in the ZK, not in `protocolVersionToReader`, because we have a fall back mechanism to read from ZK if local `protocolVersionToReader` doesn't have it, I should have waited for it to appear in HelixReadWriteSchemaRepository instead of  `protocolVersionToReader`, does that make sense?

Reply:
IIRC, you're referring to the following logic. After reading from ZK, it also puts the schema into `protocolVersionToReader` in `cacheDatumReader`. But I didn't get why waiting this fallback ensures that local protocol loading completes. I think the loading of **local** protocols finishes during the construction time. The fallback is triggered when one consumer does not have the schema that another producer has, which is not the case right after the setup of this integ test.
```
Schema newProtocolSchema = schemaReader.getValueSchema(protocolVersion);
if (newProtocolSchema == null) {
    throw new VeniceMessageException(
        "Received Protocol Version '" + protocolVersion + "' which is not currently known by " +
        this.getClass().getSimpleName() + ". A remote fetch was attempted, but the " +
        SchemaReader.class.getSimpleName() + " returned null" +
        ". The currently known Protocol Versions are: " + getCurrentlyLoadedProtocolVersions() + ".");
}
specificDatumReader = cacheDatumReader(protocolVersion, newProtocolSchema);
```

Waiting for the schema to appear in ZK is still beneficial though, which could help avoid the race that the latest schema is removed but then added back by leader controller initializer.

Reply:
Yes, you're right and it doesn't make sense to wait for the local ones stored in `protocolVersionToReader` and it doesn't make sense.
> I think the loading of local protocols finishes during the construction time.

This is exactly what I wanted to wait for and why. It's a mistake in the previous version to wait for a different one. :-) 
> Waiting for the schema to appear in ZK is still beneficial though, which could help avoid the race that the latest schema is removed but then added back by leader controller initializer.